source/dofs/dof_tools.cc

```diff
@@ -765,6 +765,51 @@ namespace DoFTools
 
 
 
+  template <typename DoFHandlerType>
+  IndexSet
+  extract_dofs_with_support_on(const DoFHandlerType &dof_handler,
+                               const std_cxx11::function< bool(const typename DoFHandlerType::active_cell_iterator &)> &predicate)
+  {
+    IndexSet support_set(dof_handler.n_dofs());
+    // first, get all the dofs which live on the subdomain:
+    std::vector<types::global_dof_index> local_dof_indices;
+    local_dof_indices.reserve (max_dofs_per_cell(dof_handler));
+
+
+    typename DoFHandlerType::active_cell_iterator
+    cell = dof_handler.begin_active(),
+    endc = dof_handler.end();
+    for (; cell!=endc; ++cell)
+      if (predicate(cell))
```

Comment:
I suspect that you want to exclude artificial cells here. It's true that the predicate could do that, but nothing good can come of it if the predicate does not do that because you will get an exception from `get_dof_indices()` a few lines further down.